### ISSUE TITLE
Can't switch to Jupyter-PowerShell kernel.  Only in web browser

### ISSUE THREAD

**Issue by mkanet** [microsoft/vscode-jupyter#4198]
## Environment data

-   VS Code version: 1.51.1
-   Jupyter Extension version (available under the Extensions sidebar):  2020.12.411183155
-   Python Extension version (available under the Extensions sidebar):  2020.11.381526539
-   OS (Windows | Mac | Linux distro) and version: Windows Version 2004
-   Python and/or Anaconda version: Python 3.7.4 (64bit)
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
-   Jupyter server running: Local | Remote | N/A

I am using [jupyter-powershell](https://github.com/vors/jupyter-powershell).  **I don't have any issues creating a new PowerShell notebook using Jupyter Notebook in a web browser.** 

Steps to recreate the issue:
1. In VSCode, open command palette
2. Select Jupyter: Create New Blank Jupyter Notebook
3. Switch to PowerShell Kernel
![PowerShellKernelSelect](https://user-images.githubusercontent.com/811008/101731941-3c297f00-3a71-11eb-83b3-bb7a4659ee58.png)
4. VSCode shows the notification:  "Switching kernel to 'PowerShell' for several seconds then the notification just goes away without actually switching the kernel.

Could someone please help get this working?  I have no idea how to debug something like this; especially when there are no errors.  I really would like to use VSCode instead of my web browser.

Thank you so much in advance!

**Comment by DonJayamanne** [comment]
Thanks for replying this issue.
How did you install the powershell kernel?

Please charge your logging to debug (go into settings and change the setting `Jupyter->ligging` to debug).
Next try once again and then please share your logs (output from the Jupyter output panel)

**Comment by mkanet** [comment]
Hi @DonJayamanne I installed the powershell kernel with two commands (per their instructions):
```
pip install powershell_kernel
python -m powershell_kernel.install
```

When I change the kernel from `Python` to `PowerShell`, the debug log is below:

```
Started kernel PowerShell
Info 2020-12-10 12:57:16: Cached data exists getEnvironmentVariables, d:\OneDrive\Command line\Untitled-1.ipynb
Info 2020-12-10 12:57:16: No custom variables for Kernel as interpreter path is not defined for kernel
Info 2020-12-10 12:57:16: > ~\AppData\Local\Programs\Python\Python37\python.exe -m powershell_kernel -f ~\AppData\Local\Temp\tmp-93645Vd3y0B8ifU0.json
> ~\AppData\Local\Programs\Python\Python37\python.exe -m powershell_kernel -f ~\AppData\Local\Temp\tmp-93645Vd3y0B8ifU0.json
Info 2020-12-10 12:57:16: cwd: d:\OneDrive\Command line
cwd: d:\OneDrive\Command line
Info 2020-12-10 12:57:16: KernelProcess error: Traceback (most recent call last):
  File "C:\Users\micha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main

Warn 2020-12-10 12:57:16: StdErr from Kernel Process Traceback (most recent call last):
  File "C:\Users\micha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main

Info 2020-12-10 12:57:16: KernelProcess error:     "__main__", mod_spec)
  File "C:\Users\micha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\micha\AppData\Local\Programs\Python\Python37\lib\site-packages\powershell_kernel\__main__.py", line 1, in <module>
    from ipykernel.kernelapp import IPKernelApp
ModuleNotFoundError: No module named 'ipykernel'

Warn 2020-12-10 12:57:16: StdErr from Kernel Process     "__main__", mod_spec)
  File "C:\Users\micha\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\micha\AppData\Local\Programs\Python\Python37\lib\site-packages\powershell_kernel\__main__.py", line 1, in <module>
    from ipykernel.kernelapp import IPKernelApp
ModuleNotFoundError: No module named 'ipykernel'

Info 2020-12-10 12:57:16: KernelProcess Exit Exit - 1
Error 2020-12-10 12:57:46: Timed out waiting to get a heartbeat from kernel process.
```

**Comment by DonJayamanne** [comment]
Thanks for the details.
Looks like we're treating the powershell kernel as a python kernel (see errors).

**Comment by mkanet** [comment]
Yes.  I noticed it's trying to load the 'ipykernel' module even though it's trying to use the PowerShell kernel.  It would be great if there's a temporary solution/workaround until there's an official fix.

**Comment by DonJayamanne** [comment]
A temporary work around would be:
* Install VS Code insiders
* Install .NET core 5
* Install the .NET Interactive kernel (see instructions here 
	* `dotnet tool install --global Microsoft.dotnet-interactive`
	* Finally install the kernels `dotnet interactive jupyter install`
* Now open a notebook & select `.NET Powershell` core kernel by `dotnet`

Sorry, but this is a work around until we fix this issue.

**Comment by mkanet** [comment]
Thank you.  Installing all those components is an overkill.  Do you have a rough estimate when this will be fixed in an upcoming Insider update?  I think I can wait.

**Comment by DonJayamanne** [comment]
Apologies for not getting back to you. Unfortunately I cannot provide an ETA for this, we'll need to triage this and priortize this with the rest of the bugs.

**Comment by greazer** [comment]
@mkanet, while we do not block the behavior, we do not officially support any non-python kernels for notebooks when using the stable build of VS Code. However, we are officially adding support for such kernels in our notebook support update currently shipping in the insiders build of VS Code. 

Therefore, the aspect of installing VS Code insiders in the workaround that Don mentions above is going to be necessary. When this support will make it to VS Code stable, is not yet defined.

**Comment by mkanet** [comment]
Thanks @greazer.  I guess I'll just have to wait for the stable version to try the workaround.  

In the meantime, I'll just use regular Jupyter Notebooks, since it doesn't require a proprietary version of non-python Kernel kernels.  In fact, I also use a PHP kernel in Jupyter Notebooks which also works great in Jupyter Notebooks, just not in VSCode's Jupyter Notebook implementation.

**Comment by DonJayamanne** [comment]
@mkanet 
Please could you provide the following info (links), so we can test these kernel at our end when we address this issue:
* How to install the powershell kernel
* How to install the php kernel

Thanks

**Comment by mkanet** [comment]
@DonJayamanne thank you for looking into it.  It sure would be nice to have the same flexibility in VSCode as the actual Jupyter Notebooks (regardless of which Kernel is used).

**Follow the instructions here:**
https://litipk.github.io/Jupyter-PHP-Installer/

I had some trouble installing the Kernel under Windows 10 since some of the dependencies were missing..  Below are some supplemental notes for the missing dependencies in my case:

My PHP installation required for me to download the compiled version of `libzmq.dll` from the web and place it in my PHP install folder. ...As well as the right version of the `php_zmq.dll` extension in in my PHP extensions folder; plus, I had to copy `api-ms-win-crt-heap-l1-1-0.dll` and `api-ms-win-crt-runtime-l1-1-0.dll` from another folder for my C+ runtime install to my PHP install folder. After I did all this, I was finally able to install **Composer** as well as **Jupyter-PHP**.
